warehouse/ingest-core/src/main/java/datawave/ingest/config/TableConfigCache.java

```
@@ -48,17 +49,55 @@ public void update() {
     @Override
     public void writeCacheFile(FileSystem fs, Path tmpCacheFile) {
         try (PrintStream out = new PrintStream(new BufferedOutputStream(fs.create(tmpCacheFile)), false, "UTF-8")) {
-            for (Map.Entry e : configMap.entrySet()) {
-                out.println(e.getKey() + DELIMITER + e.getValue());
+            for (Map.Entry<String,Map<String,String>> table : configMap.entrySet()) {
+                for (Map.Entry tableProp : table.getValue().entrySet()) {
+                    out.println(table.getKey() + DELIMITER + tableProp.getKey() + DELIMITER + tableProp.getValue());
+                }
             }
         } catch (IOException e) {
             log.error("Unable to write cache file " + tmpCacheFile, e);
         }
     }
     
+    @Override
+    protected void readCache(BufferedReader in, String delimiter) throws IOException {
+        this.configMap = new HashMap<>();
+        String line;
+        String table = null;
+        String propName;
+        String propVal;
+        Map<String,String> tempMap = new HashMap<>();
+        while ((line = in.readLine()) != null) {
+            String[] parts = StringUtils.split(line, delimiter);
+            if (table == null || !table.equals(parts[0])) {
+                table = parts[0];
+                tempMap = new HashMap<>();
+                this.configMap.put(table, tempMap);
+            }
+            if (parts.length == 3) {
+                propName = parts[1];
+                propVal = parts[2];
+                tempMap.put(propName, propVal);
+            }
```

Perhaps we should be failing if the file does not match our format as that would imply its the wrong file or something failed silently while writing.

---

Sure